guardian/facia-scala-client
# Introduce ContentFormat into FAPI Client

## What does this change?


Both Apps and Dotcom platforms currently use a `Format` type to determine how to style content. This type is defined in the [`@guardian/types` library](https://github.com/guardian/types/blob/main/src/format.ts) and, as of a recent PR, [the Scala CAPI Client](https://github.com/guardian/content-api-scala-client/blob/b8dc1c48c845bdc824a699fea2e31286d0cf2eba/client/src/main/scala/com.gu.contentapi.client/utils/CapiModelEnrichment.scala#L69).

In order for this information to be used in Trails and on Cards the FAPI client needs to be updated to use the new information from the Scala client and pass it through to the consumers.

I here add the `Format` type to the `CuratedContent`, `SupportingCuratedContent` and `LatestSnap` classes to allow it to be consumed on-platform.

### Changes:
- Bump CAPI client version to the most recent version as of writing (v17.11)
- Add a `format` field to the above listed classes and create a `ContentFormat` case class to type this field.
  - This case class is copied over from Frontend, but there is future work to move both `Format` and its types to the top level rather than in the client. At this point the duplication will no longer be necessary.

### Related PRs:
 - Introduction of `Format` to the CAPI client: https://github.com/guardian/content-api-scala-client/pull/312
 - Implementation of `Format` in Frontend: https://github.com/guardian/frontend/pull/23577

## How to test

Integrate new version into consumers and confirm it works (this feels a bit circular, but I can't see a better way to test this).

## How can we measure success?

Ability to add Format data to trails on platforms.

## Have we considered potential risks?

This is additive and so shouldn't break existing content, but anyone bumping version might need to update their types. Compilation should catch this however
